quinn-proto/src/tests/mod.rs

```diff
@@ -666,6 +666,114 @@ fn zero_rtt_rejection() {
     assert_eq!(pair.client_conn_mut(client_ch).lost_packets(), 0);
 }
 
+fn test_zero_rtt_incoming_limit<F: FnOnce(&mut ServerConfig)>(configure_server: F) {
+    // caller sets the server limit to 4000 bytes
+    // the client writes 8000 bytes
+    const CLIENT_WRITES: usize = 8000;
+    // this gets split across 8 packets
+    // the first packet is stored in the Incoming
+    // the next three are incoming-buffered, bringing the incoming buffer size to 3600 bytes
+    // the last four are dropped due to the buffering limit and must be retransmitted
+    const EXPECTED_DROPPED: u64 = 4;
+
+    let _guard = subscribe();
+    let mut server_config = server_config();
+    configure_server(&mut server_config);
+    let mut pair = Pair::new(Arc::new(EndpointConfig::default()), server_config);
+    pair.server.incoming_connection_behavior = IncomingConnectionBehavior::Validate;
```

Comment:
Do we need to be overriding the default here?

Reply:
I'm not sure why it would be important. It was just copied from the `zero_rtt_happypath` test, so I'd be curious why it was originally added there.

Reply:
Digging through history, it looks like it was added in 3b01662324a14b605f5e7f54922c445212835b9f, to give test coverage for the case where the client must retransmit 0-RTT when responding to a Retry during a 0-RTT connection attempt. I think this is valuable for the happypath test, but unnecessary here, especially since using `Wait` for the second (0-RTT) connection attempt undermines it anyway.

Reply:
Removed.